### PR TITLE
Support sharded exports to enable replications of larger tables

### PR DESCRIPTION
**Pull request by courtsVII**

### CODE REVIEW

**Comment by courtsVII** [comment]
fixes #10

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java [hunk 1]
```diff
@@ -44,20 +44,41 @@ void extract(Table table, BigQueryExtractionData extractionData) {
...
+  private void deleteBucketAndContents(String dataBucket) {
+    deleteObjectsInBucket(dataBucket);
+    deleteBucket(dataBucket);
+  }
 
-    BlobId blobId = BlobId.of(dataBucket, dataKey + "." + format);
-    boolean suceeded = storage.delete(blobId);
-    if (!suceeded) {
-      log.warn("Could not delete object {}", dataUri);
+  private void deleteBucket(String dataBucket) {
```
Comment (massdosage) [inline src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: Swap this method and the next one around, makes it easier to read if they are in the order they are executed as per method above.

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java [hunk 2]
```diff
@@ -44,20 +44,41 @@ void extract(Table table, BigQueryExtractionData extractionData) {
...
+      for (Blob blob : blobs) {
+        boolean suceeded = storage.delete(blob.getBlobId());
+        if (suceeded) {
+          log.info("Deleted object {}", blob);
+        } else {
+          log.warn("Could not delete object {}", blob);
+        }
+      }
+    } catch (Exception e) {
+      log.warn(e.toString());
```
Comment (massdosage) [inline src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: Shouldn't we log the exception too? i.e. `log.warn("Error deleting objects in bucket {}", bucket, e)`.

File: src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java [hunk 3]
```diff
@@ -98,6 +109,12 @@ public void cleanupWhenDeletionFailsDoesntThrowException() {
     Bucket bucket = mock(Bucket.class);
     when(bucket.delete()).thenReturn(true);
     when(storage.get(anyString())).thenReturn(bucket);
+    Blob blob = mock(Blob.class);
+    List<Blob> blobs = ImmutableList.of(blob);
```
Comment (massdosage) [inline src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java]: If want to the test to be a bit more realistic we should have more than one blob in the list and then verify that delete was called for all of them.

**Comment by massdosage** [comment]
Above changes look good.

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java [hunk 4]
```diff
@@ -44,20 +44,41 @@ void extract(Table table, BigQueryExtractionData extractionData) {
...
+      Iterable<Blob> blobs = storage.list(dataBucket).iterateAll();
+      for (Blob blob : blobs) {
+        boolean suceeded = storage.delete(blob.getBlobId());
+        if (suceeded) {
+          log.info("Deleted object {}", blob);
+        } else {
+          log.warn("Could not delete object {}", blob);
+        }
+      }
+    } catch (Exception e) {
```
Comment (patduin) [inline src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: can't we catch anything more specific? 
Should we move the try-catch inside the for loop so around every delete, so all deletes are attempted?

File: src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java [hunk 5]
```diff
@@ -84,8 +90,26 @@ public void cleanup() {
     Bucket bucket = mock(Bucket.class);
```
Comment (patduin) [inline src/test/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionServiceTest.java]: Add tests for the Exception handling

File: src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java (hunk 4 above)
Comment (massdosage) [reply src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/BigQueryDataExtractionService.java]: Above would be nice, @courtsVII when you have a chance and if the above look viable I'd suggest either a small PR with just this change, or alternatively a ticket describing this so we can do it later.

**Pull request merged**